resources/assets/js/search.js

```diff
@@ -0,0 +1,147 @@
+    }
+
+    function parseResults(data) {
+        let results = [];
+        data.forEach(function (contact) {
+            let person = {};
+            person.url = `/people/${contact.id}`;
+
+            const middleName = contact.middle_name || '';
+            const lastName = contact.last_name || '';
+            // Unify first, middle and last name in one string (depending on availability).
+            person.name = contact.first_name + (middleName ? ' ' + middleName : '') + (lastName ? ' ' + lastName : '');
+
+            // Figure out which avatar to use and create the appropriate HTML.
+            person.avatar = getAvatar(contact);
+
+            results.push(person);
+        });
+
+        return results;
+    }
+
+    function getAvatar(contact) {
+        let avatar;
+        if ((contact.has_avatar === 'true')) {
+            avatar = `<img src="/storage/${contact.avatar_file_name}" class="avatar">`;
+        } else if (contact.gravatar_url !== null) {
+            avatar = `<img src="${contact.gravatar_url}" class="avatar">`;
+        } else {
+            let initials = contact.first_name.substring(0, 1);
+            initials += contact.middle_name ? contact.middle_name.substring(0, 1) : '';
+            initials += contact.last_name ? contact.last_name.substring(0, 1) : '';
+            initials = initials.toUpperCase();
+            avatar = `<div class="avatar avatar-initials" style="background-color: ${contact.default_avatar_color}">${initials}</div>`;
+        }
+
+        return avatar;
+    }
+
+    function getInputValue() {
+        let value = search.input.eq(0).val();
+        if (value === '') {
+            value = search.input.eq(1).val();
+        }
+        return value;
+    }
+
+    function searchInContacts() {
+        const needle = getInputValue();
+        if(needle === '') {
+            return;
+        }
+
+        $.post({
+            url: "/people/search",
+            data: {
+                needle: needle
+            }
+        }).done(function (data) {
+            if (data.noResults !== undefined) {
+                showNoResults(data.noResults);
+                return;
+            }
+            const results = parseResults(data);
+            showResults(results);
+        });
+    }
+
+    /* -----------------------------------------------------------------------------------
+     | Event binding.
+     |---------------------------------------------------------------------------------- */
+    function bindEvents() {
+        // We want typeahead-like behaviour, but not a query for every keystroke.
+        // We submit a query 200ms after a key has been pressed.
+        search.input.on('keyup', function () {
+            if (search.timeoutId !== undefined) {
+                window.clearTimeout(search.timeoutId);
+            }
+            search.timeoutId = window.setTimeout(searchInContacts, 200);
+        });
+
+        search.input.on('submit', function (e) {
```

Comment:
Can you add a check here? At the moment, if we enter something and press Enter, the Search will be triggered and we'll receive an error page.